### DISCUSSION TITLE
SQLITE JSON comparison

### DISCUSSION THREAD

**Discussion by alex-eri** [sequelize/sequelize#15654] [Help & Questions]
My setup simplified

```
export const sequelize = new Sequelize("sqlite:database.sqlite");

export const Os = sequelize.define("Os", {
  v: DataTypes.JSON,   
});
```

`await Os.upsert({v:2})`  inserted as `"2"` instead of `2`

`await Os.upsert({v:"2"})`  inserted as `"2"`

`Os.count({ where: { v :  "2"}})` evaluated to `WHERE Os.v = '2'` - not matched

`Os.count({ where: { v : sequelize.fn('json', "2")}})` evaluated to `WHERE Os.v = JSON('2')` - not matched

`Os.count({ where: { v : JSON.stringify("2")}})`  evaluated to `WHERE Os.v = JSON('"\"2\""')` - not matched

And super ugly workaround:

`Os.count({ where: { v: sequelize.fn('json', JSON.stringify("2"))` - matches !!!

**Comment by ephys** [comment]
I'm not sure I understand, but how JSON works has been rewritten in v7 in https://github.com/sequelize/sequelize/pull/14505 & https://github.com/sequelize/sequelize/pull/15598, if you're interested in testing that it produces the results you expect
